projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-row-dimension-header.component.ts

```diff
@@ -0,0 +1,31 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, HostListener, Inject } from '@angular/core';
+
+import { GridType, IGX_GRID_BASE } from '../common/grid.interface';
+
+import { IgxGridHeaderComponent } from '../headers/grid-header.component';
+import { IgxColumnResizingService } from '../resizing/resizing.service';
+
+/**
+ * @hidden
+ */
+@Component({
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    selector: 'igx-pivot-row-dimension-header',
+    templateUrl: '../headers/grid-header.component.html'
+})
+export class IgxPivotRowDimensionHeaderComponent extends IgxGridHeaderComponent {
+
+    constructor(
+        @Inject(IGX_GRID_BASE) public grid: GridType,
+        public colResizingService: IgxColumnResizingService,
+        public cdr: ChangeDetectorRef,
+        private refInstance: ElementRef<HTMLElement>
+    ) {
+        super(grid, colResizingService, cdr, refInstance);
+    }
+
+    @HostListener('click', ['$event'])
+    public onClick(event: MouseEvent) {
```

Comment:
@mddragnev if we extend this we can also remove the pivot specific check in `IgxGridHeaderComponent`'s `onClick` method. We should revert the change from:
```
        if (!this.grid.isPivot || !this.grid.navigation.isRowHeaderActive) {
            this.grid.theadRow.nativeElement.focus();
        }
```
to:

```
    this.grid.theadRow.nativeElement.focus();
```